File: src/main/java/com/bajdcc/LALR1/syntax/automata/nga/NGABag.java
package com.bajdcc.LALR1.syntax.automata.nga;

import com.bajdcc.LALR1.syntax.ISyntaxComponent;

import java.util.ArrayList;
import java.util.Stack;

/**
 * 非确定性文法自动机结果包
 *
 * @author bajdcc
 */
public class NGABag {
	/**
	 * NGA栈
	 */
	public Stack<ArrayList<ENGA>> stkNGA = new Stack<>();

	/**
	 * NGA子表
	 */
	public ArrayList<ENGA> childNGA = new ArrayList<>();

	/**
	 * 存储结果的ENGA
	 */
	public ENGA nga = null;

	/**
	 * 标记前缀
	 */
	public String prefix = "";

	/**
	 * 标记前缀
	 */
	public ISyntaxComponent expression = null;
}

File: src/main/java/com/bajdcc/LALR1/syntax/automata/nga/NGABag.kt
package com.bajdcc.LALR1.syntax.automata.nga

import com.bajdcc.LALR1.syntax.ISyntaxComponent
import java.util.*

/**
 * 非确定性文法自动机结果包
 * [stkNGA] NGA栈
 * [childNGA] NGA子表
 * [nga] 存储结果的ENGA
 * [prefix] 标记前缀
 * [expression] 表达式
 * @author bajdcc
 */
data class NGABag(var stkNGA: Stack<MutableList<ENGA>> = Stack(),
                  var childNGA: MutableList<ENGA> = mutableListOf(),
                  var nga: ENGA? = null,
                  var prefix: String = "",
                  var expression: ISyntaxComponent? = null)